File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/db2/BinaryTypes.kt
package dev.typr.foundations.docs.db2

import dev.typr.foundations.Db2Type
import dev.typr.foundations.Db2Types

@Suppress("unused")
class BinaryTypes {
    //start
    val binaryType: Db2Type<ByteArray> = Db2Types.binary
    val binary16: Db2Type<ByteArray> = Db2Types.binary(16)
    val varbinaryType: Db2Type<ByteArray> = Db2Types.varbinary
    val blobType: Db2Type<ByteArray> = Db2Types.blob
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/mariadb/BinaryTypes.scala
package dev.typr.foundationssc.docs.mariadb
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

@SuppressWarnings(Array("unused"))
object BinaryTypes:
  // start
  val binaryType: MariaType[Array[Byte]] = MariaTypes.binary(16)
  val varbinaryType: MariaType[Array[Byte]] = MariaTypes.varbinary(255)
  val blobType: MariaType[Array[Byte]] = MariaTypes.blob
  // stop
